sdk/servicebus/service-bus/src/core/managementClient.ts

```diff
@@ -475,6 +476,13 @@ export class ManagementClient extends LinkEntity {
         const wrappedEntry = types.wrap_map(entry);
         messageBody.push(wrappedEntry);
       } catch (err) {
+        if (err instanceof TypeError || err.name === "TypeError") {
```

Comment:
One or the other checks here is probably sufficient?

Reply:
We need both.
- `err instanceOf TypeError` catches errors that nodejs throws when it finds type errors when running rhea
- `err.name === "TypeError"` catches errors that [rhea explicitly throws](https://github.com/amqp/rhea/blob/df00b8c187d704a339044c793ff5cd24a2e259fc/lib/errors.js#L27) type errors, but actually throws normal errors with the name `TypeError`.

Will add a comment

Reply:
Comment added: https://github.com/Azure/azure-sdk-for-js/pull/2536/commits/4611fc816949aacf65a9a5a15e1d26d8fc8ac2fd